integration_tests/opentsdb/test-put.py

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import requests
+import time
+
+api_root = 'http://localhost:5440'
+headers = {
+    'Content-Type': 'application/json'
+}
+table_prefix = 'opentsdb_test_'
+table2_prefix = 'opentsdb_test2_'
+
+
+def now():
+    return int(time.time()) * 1000
+
+
+def execute_sql(sql):
+    r = requests.post('{}/sql'.format(api_root), json={'query': sql}, headers=headers)
+    return r
+
+def drop_table(table_name):
+    sql = """
+DROP TABLE {}    
+    """.format(table_name)
+    r = execute_sql(sql)
+    # 200 for success, 500 for table not found.
+    assert r.status_code == 200 or r.status_code == 500, r.text
+
+
+def show_create_table(table_name):
+    sql = """
+SHOW CREATE TABLE {}    
+    """.format(table_name)
+    r = execute_sql(sql)
+    assert r.status_code == 200
+    return r
+
+
+def execute_sql_query(sql):
+    r = execute_sql(sql)
+    assert r.status_code == 200
+    return r
+
+
+def execute_put(points):
+    r = requests.post('{}/opentsdb/api/put'.format(api_root), data=points)
+    return r
+
+
+def execute_put_then_assert_fail(points):
+    r = execute_put(points)
+    assert r.status_code == 500
+
+
+def execute_put_then_assert_success(points):
+    r = execute_put(points)
+    assert r.status_code == 204
+
+
+def test_put_validate_error():
+    ts = now();
+    # empty json string
+    execute_put_then_assert_fail("")
+
+    # invalid json
+    execute_put_then_assert_fail("{xxx")
+
+    # empty metric
+    execute_put_then_assert_fail("""
+{
+    "metric": "",
+    "timestamp": {ts},
+    "value": 18,
+    "tags": {
+        "host": "web01",
+        "dc": "lga"
+    }
+}
+    """.replace('{ts}', str(ts)))
+
+    # empty tag
+    execute_put_then_assert_fail("""
+{
+    "metric": "sys.cpu.nice",
+    "timestamp": {ts},
+    "value": 18,
+    "tags": {
+    }
+}
+    """.replace('{ts}', str(ts)))
```

Comment:
```
print('Hello {name}!This is {program}.'.format(name='foo',program='john'))
```
format already support name replacement.

Reply:
The template is a json string which contains many '{' and '}' characters, so it's not convenient to use `format`